cou/utils/juju_utils.py

```diff
@@ -699,6 +699,7 @@ async def _wait_for_idle() -> None:
                             raise_on_blocked=raise_on_blocked,
                             raise_on_error=raise_on_error,
                             status=status,
+                            wait_for_at_least_units=0,  # don't hang if an app has no units
```

Comment:
I wonder if we have been doing this erroneously: by default this is waiting for only 1 unit, so if we have more than one units for this `app`, then only one unit is in idle is enough

Reply:
So I think checking for number of units is independent of the status check. The logic seems to be:

1. check number of units
2. for each unit:
  3. check unit status